src/window.py

```diff
@@ -471,9 +475,22 @@ def _on_load_finished(self, worker, directory):
         self._go_to_top()
         self._update_all()
 
-    def _on_load_failed(self, worker, directory):
+    def _on_load_failed(self, worker, path, ex):
+
+
+        self._busy = False
         self._clean_workers()
 
+        self.loading_label.set_text(_("Error loading %s") % path)
+        self.loading_description.set_text(_("Cause: %s") % str(ex))
+
+        self.action_stack.set_visible_child(self.close_box)
+        self.tools_stack.set_visible_child(self.close_tools)
+
+        # Inject GO_HOME on the callback of the close method
+        self.close_button.handler_block(self.handler_close)
+        self.close_button.handler_unblock(self.handler_close_go_home)
```

Comment:
Here is where the two lines I wrote earlier should go, and no need to keep the handler.. because..